velocity_modelling/cvm/velocity3d.py

```diff
@@ -0,0 +1,786 @@
+                    depth_subset,
+                    self,
+                    mesh_vector,
+                    nz_tomography_data,
+                    partial_global_surface_depths,
+                    global_params["GTL"],
+                    in_any_basin_lat_lon,
+                    on_boundary,
+                    interpolated_global_surface_values,
+                )
+            elif name == "canterbury1d_v1":
+                from velocity_modelling.cvm.submodel import canterbury1d_v1
+
+                canterbury1d_v1.main_vectorized(
+                    index_subset, depth_subset, self, velo_mod_1d_data
+                )
+            else:
+                raise ValueError(f"Error: Submodel {name} not found in registry.")
+
+    # TODO: I need to fix this function to automatically trigger the right main function of the submodel
+    # some unused positional arguments are here for future use. will think of a better way to handle this
+    def call_basin_submodel_vectorized(
+        self,
+        partial_basin_surface_depths: PartialBasinSurfaceDepths,
+        partial_global_surface_depths: PartialGlobalSurfaceDepths,
+        nz_tomography_data: TomographyData,
+        mesh_vector: MeshVector,
+        in_any_basin_lat_lon: bool,
+        on_boundary: bool,
+        depths: np.ndarray,
+        ind_above: int,
+        basin_num: int,
+        z_indices: np.ndarray,
+    ):
+        """
+        Call the appropriate basin sub-velocity models for multiple depths.
+
+        Parameters
+        ----------
+        partial_basin_surface_depths : PartialBasinSurfaceDepths
+            Basin surface depths at this location.
+        partial_global_surface_depths : PartialGlobalSurfaceDepths
+            Global surface depths at this location.
+        nz_tomography_data : TomographyData
+            Tomography data.
+        mesh_vector : MeshVector
+            Lat-lon point with multiple depths.
+        in_any_basin_lat_lon : bool
+            Whether this lat-lon point is in any basin.
+        on_boundary : bool
+            Whether this point is on a model boundary.
+        depths : np.ndarray
+            Depths (m) to calculate properties for.
+        ind_above : int
+            Index of surface above these depths.
+        basin_num : int
+            Basin identifier.
+        z_indices : np.ndarray
+            Indices of depths in the mesh.
+        """
+
+        basin_data = partial_basin_surface_depths.basin
+        submodel_name, submodel_data = basin_data.submodels[ind_above]
+
+        if submodel_name == "nan_submod":
+            from velocity_modelling.cvm.submodel import nan_submod
+
+            nan_submod.main_vectorized(z_indices, self)
+
+        elif submodel_name in [
+            "canterbury1d_v1",
+            "canterbury1d_v2",
+            "canterbury1d_v2_pliocene_enforced",
+        ]:
+            from velocity_modelling.cvm.submodel import canterbury1d_v1
```

Comment:
Could these imports be at the top too?

Reply:
Ideally, I want the import to be done automatically/dynamically from the submodel names, so that a new submodel can be added to the platform like a plugin without changing the code. I was running out of an idea for an elegant solution. I'd love a suggestion!